### PR TITLE
Enabled print option and updated the object types

### PR DESCRIPTION
**Pull request by anastasia-mbithe**
This fixes (parts a to d) #8163 

After a discussion with @Patowhiz, we settled to use `get_object_data `function to print the objects since the print function is not yet implemented to support `html `object types.

Comment e) is yet to be implemented.

### CODE REVIEW

**Comment by lloyddewit** [comment]
> Comment e) is yet to be implemented.

@anastasia-mbithe Thank you for this PR.
If part e is not yet implemented then will you create a new issue for part e? Or will you update the comment to 'fixes (parts a to d) of #8163'?
Otherwise the issue will close completely when this PR is merged.
thanks

**Comment by N-thony** [comment]
@derekagorhom can you peer review this?

**Review by rdstern — COMMENTED** [review]
@anastasia-mbithe this looks excellent.
The one item to change in the view is (I think) here:
image

When I return to the View dialogue it has kept the last object.  You can see here I had changed the name, but it kept the old name.
It also keeps the name when you change the type of object you are having the structure or printing.  

I am happy that the name remains when you return to the View dialogue.  But if you change the object type, then it should clear.  If you rename the selected object, it should either clear the old name or swap it for the new name.  

@Patowhiz anything more?  It would be great to merge this improvement.

**Comment by anastasia-mbithe** [comment]
@rdstern, The issue hasn't been fixed yet. It's what am working on with @N-thony.

**Comment by anastasia-mbithe** [comment]
@rdstern This is ready for testing

File: instat/dlgDeleteObjects.vb [hunk 1]
```diff
@@ -53,12 +53,13 @@ Public Class dlgDeleteObjects
...
-        dctTypes.Add("Filters", Chr(34) & "filter" & Chr(34))
-        dctTypes.Add("Column selections", Chr(34) & "column_selection" & Chr(34))
-        dctTypes.Add("Calculations", Chr(34) & "calculation" & Chr(34))
+        dctTypes.Add("Summaries", Chr(34) & "summary" & Chr(34))
         dctTypes.Add("Tables", Chr(34) & "table" & Chr(34))
         dctTypes.Add("Graphs", Chr(34) & "graph" & Chr(34))
         dctTypes.Add("Models", Chr(34) & "model" & Chr(34))
+        dctTypes.Add("Filters", Chr(34) & "filter" & Chr(34))
+        dctTypes.Add("Column selections", Chr(34) & "column_selection" & Chr(34))
+        dctTypes.Add("Calculations", Chr(34) & "calculation" & Chr(34))
```
Comment (Patowhiz) [inline instat/dlgDeleteObjects.vb:62]: @anastasia-mbithe are calculations output objects? If so then they I think we should classify them as summaries or structured depending on what they really are.

File: instat/dlgViewObjects.vb [hunk 2]
```diff
@@ -40,53 +41,68 @@ Public Class dlgViewObjects
...
+        ucrPnlContentsToView.AddParameterValuesCondition(rdoPrint, "check", "print")
+        ucrPnlContentsToView.AddParameterValuesCondition(rdoStructure, "check", "str")
+
+        ucrInputObjectType.SetParameter(New RParameter("object_type", 0))
+        dctTypes.Add("Objects", Chr(34) & "object" & Chr(34))
+        dctTypes.Add("Summaries", RObjectTypeLabel.Summary)
+        dctTypes.Add("Tables", RObjectTypeLabel.Table)
+        dctTypes.Add("Graphs", RObjectTypeLabel.Graph)
+        dctTypes.Add("Models", RObjectTypeLabel.Model)
+        ucrInputObjectType.SetItems(dctTypes)
```
Comment (Patowhiz) [inline instat/dlgViewObjects.vb:72]: @anastasia-mbithe I think structured is missing here? the survival dialogs can produce structured? I'm happy we add this later.

File: instat/dlgViewObjects.vb [hunk 3]
```diff
@@ -112,5 +132,16 @@ Public Class dlgViewObjects
...
     End Sub
 
+    Private Sub ucrInputObjectType_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrInputObjectType.ControlValueChanged
+        Dim key As String = dctTypes.Keys(ucrInputObjectType.cboInput.SelectedIndex)
+        Dim value As String = ""
 
+        If Not ucrReceiverSelectedObject.IsEmpty AndAlso ucrSelectorForViewObject.lstAvailableVariable.FindItemWithText(ucrReceiverSelectedObject.GetVariableNames(False), True, 0, False) Is Nothing Then
+            ucrReceiverSelectedObject.SetText("")
+        End If
+        If key IsNot Nothing AndAlso dctTypes.TryGetValue(key, value) Then
```
Comment (Patowhiz) [inline instat/dlgViewObjects.vb:142]: I suggest using `dctTypes.contains()` it would simplify the code.

**Review by Patowhiz — COMMENTED** [review]
@anastasia-mbithe kindly test the following;
1. Produce a graph object
2. Use the print option in this dialog to check if the R-viewer will be shown.
3. Use the R-viewer in the last graph option in the main form tool bar. Check if R-viewer is shown.

Do steps 2 and 3 in 3 different orders.
Is the R-viewer always shown for graphs? What's the difference in the R scripts produced.

**Comment by anastasia-mbithe** [comment]
For the first comment,
@Patowhiz I have done this severally, lemme use the last trial; So I made two graphs, last_graph, and bar_plot. Viewed the last_graph via dlgViewObject Print option, then went to RViewer, and the bar_plot was viewed (it was the last graph produced from the bar chart dialog)
The difference in the codes is that the RViewer uses the `get_last_object_data` function while the dlgViewObject print option uses the `get_object_data` function.
image

**Comment by lloyddewit** [comment]
@Patowhiz Are all your comments resolved now, and can we ask @rdstern to test?
thanks

**Comment by Patowhiz** [comment]
Yes @lloyddewit. Just some minor code changes which I think @anastasia-mbithe will be pushing after our discussions.

**Comment by anastasia-mbithe** [comment]
@Patowhiz and @lloyddewit, This is ready for testing, I had pushed the changes before Patrick's comment. Sorry.

**Comment by anastasia-mbithe** [comment]
> > @Patowhiz and @lloyddewit, This is ready for testing, I had pushed the changes before Patrick's comment. Sorry.
> 
> @anastasia-mbithe is this ready for testing by @rdstern ?

Yes

**Pull request closed without merging**